### ISSUE TITLE
tracef should not emit a newline at the end

### ISSUE THREAD

**Issue by jacereda** [aduros/wasm4#612]
There should be some way to emit a string without a newline, and looks like `tracef` is the answer if it didn't emit the final newline. See https://github.com/aduros/wasm4/pull/611

In Forth it's idiomatic to emit lines like:
```
: foo ." 1+2=" 1 2 + . cr ;
```
With the current `tracef` behaviour it's inconvenient to implement that, since it would require buffering until a `cr` is performed.

**Comment by LunaAmora** [comment]
Yeah, i would prefer to have both versions as one without a newline would facilitate printing various stuff in the same line. But when i asked why it was this way, @aduros said on discord:
"Yeah, the intent was to have a drop in replacement for console.log() for debugging, and not fully expose stdout"

So i guess it is intended to not support this :/

**Comment by jacereda** [comment]
Maybe a SYSTEM_FLAG to opt-in as I suggested in https://github.com/aduros/wasm4/pull/611#issuecomment-1440704735

**Comment by Ruulul** [comment]
it is possible tho to use ANSI control codes to print on the line above, move the cursor, clear the terminal...
https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797

came pretty handy on debugging real time information to me

**Comment by JerwuQu** [comment]
I agree that having a system flag to enable this behavior would be great for the purpose of toy languages. Perhaps `SYSTEM_TRACE_RAW`?
I'm not sure how it would be done in the web runtime though. Only two solutions I can think of are either buffering with flushing on newline, or running `console.log` on every single character.
